Frameworks/UIKit/UINibUnarchiver.mm

```diff
@@ -203,7 +203,13 @@ static id constructObject(UINibUnarchiver* self, Object* pObj) {
         id classId = pObj->classType; // objc_getClass(pObj->className);
         assert(classId != nil);
 
-        pObj->cachedId = [classId alloc];
+        // TODO: #1105 alloc without init is invalid, and not all classes support alloc-init anyway
+        // look for a proper fix to this issue, rather than just special-casing for UIFont
+        if ([classId isEqual:[UIFont class]]) {
+            pObj->cachedId = [[UIFont defaultFont] retain];
```

Comment:
we weren't before, since UIFont doesn't support NSCoding, and I wanted to maintain the existing functionality. I'll wait for dave to comment.

---
In reply to: [82631100](https://github.com/Microsoft/WinObjC/pull/1106#discussion_r82631100) [](ancestors = 82631100)